AccidentalCalculator: Fix natural accidentals not remembered within measure

chbauman opened the pull request:
Adresses and fixes #1383

Now it looks like this:
![after](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/assets/33223146/ec736385-093f-4b03-acc3-3944dad604fd)

chbauman committed:
Update AccidentalCalculator.ts

Adresses #1383

sschmidTU commented:
Hey @chbauman, thanks for the contribution, however this change causes some duplicate natural accidentals.
Minimal reproducing sample:
<img width="214" alt="image" src="https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/assets/33069673/48b8952a-c919-44e9-92c6-ebf512a85f65">

Musescore:
<img width="279" alt="image" src="https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/assets/33069673/af547099-2507-4196-b1cb-f8eca8348179">

sample:
[test_accidental_natural_repeat.zip](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/files/11547653/test_accidental_natural_repeat.zip)

There's even a "double natural" in Dichterliebe measure 9:
<img width="204" alt="image" src="https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/assets/33069673/3411216d-9438-428b-b746-8bcd502c5562">

So I suspect this is rather a workaround or in any case not taking all variables into account.
In this case it seems like your change works for extra accidentals/naturals in C major key signature but not in other key signatures like D major.

The accidental display code is clearly not optimal in OSMD and there are other issues e.g. with [courtesy accidentals](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/issues?q=is%3Aissue+courtesy), and I've had a quick look before, but it looks like the accidental code needs a slightly more holistic fix.

chbauman committed:
Added additional check

The previous accidental must not be discarded if the current alteration is also a natural.

chbauman committed:
Merge branch 'opensheetmusicdisplay:develop' into patch-1

chbauman committed:
Clear current alterations

Fixes at least some issues

sschmidTU commented:
Closing, as we already/recently fixed #1383.
Still, thank you for working on it!
